### PR TITLE
Error 404 Page Added as per mentioned Issue

### PR DESCRIPTION
**Pull request by nerdyninja**

### CODE REVIEW

**Comment by dbudwin** [comment]
Thanks for this @nerdyninja!  It appears you forgot to [fill out the template provided to you when creating a pull request](https://github.com/dbudwin/RoboHome-Web/blob/master/.github/PULL_REQUEST_TEMPLATE.md).  It is essential that it be included so this can be reviewed in a timely manner.

Can you also please add a Laravel Dusk test to make sure the 404 screen is presented on invalid URLs.

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -31,9 +31,17 @@ class Handler extends ExceptionHandler
      * @param  \Exception  $exception
      * @return void
      */
-    public function report(Exception $exception): void
+    public function report(Exception $exception): Response
```
Comment (dbudwin) [inline app/Exceptions/Handler.php]: This no longer matches the return type in the comment above.  I do not believe the changes to this class are necessary.  Could you please elaborate on this change?

File: resources/views/errors/404.blade.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<html>
+    <head>
+        <title>RoboHome | Login</title>
```
Comment (dbudwin) [inline resources/views/errors/404.blade.php]: The title shouldn't contain "Login" anymore.  Maybe "RoboHome | 404 Not Found"

File: resources/views/errors/404.blade.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        <script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha256-cCueBR6CsyA4/9szpPfrX3s49M9vUU5BgtiJj06wt/s=" crossorigin="anonymous"></script>
+        <!-- Latest compiled and minified CSS -->
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+        <!-- Latest compiled and minified JavaScript -->
+        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <style>
+          /* Error Page Inline Styles */
```
Comment (dbudwin) [inline resources/views/errors/404.blade.php]: By including and customizing Bootstrap, this no longer matches the style of website.  The issue requested that the 404 error page match the style of the website.

File: resources/views/errors/404.blade.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            display.innerHTML = document.domain;
+          }
+        </script>
+    </head>
+    <body>
+       <div class="container">
+          <div class="jumbotron">
+            <h1><i class="glyphicon glyphicon-exclamation-sign"></i></h1>
+            <h2>404 Not Found</h2>
+            <p class="lead">We couldn't find what you're looking for on <em><span id="display-domain"></span></em>.</p>
```
Comment (dbudwin) [inline resources/views/errors/404.blade.php]: I'm not sure the `<em><span id="display-domain"></span></em>` part works.  When I get to the 404 page all I see is the following:

screen shot 2017-10-04 at 18 54 46

There are several really good Laravel Dusk test that should be added.  The test should:

1. Make sure the back button exists
2. Make sure the go home button exists
3. Make sure the title of the page is correct
4. Make sure the page gets loaded for a bad URL

Given that access to this page isn't behind a login wall, it shouldn't be too hard to test, see examples in this project here: https://github.com/dbudwin/RoboHome-Web/blob/master/tests/Browser/IndexTest.php

**Comment by nerdyninja** [comment]
Okay @dbudwin I have reviewed all your requests. I will be sending a PR with one commit within next 24 hours. I will make sure I follow your requirements as mentioned here.
Apologies for the inconvenience.

**Comment by dbudwin** [comment]
No worries @nerdyninja, please let me know how I can help.  I look forward to your change!

**Comment by nerdyninja** [comment]
Hey @dbudwin i might not be able to add Laravel Dusk Test. But i will be able to solve rest of the issue.

**Pull request closed without merging**